metricbeat/mb/module/publish.go

```diff
@@ -31,14 +34,40 @@ import (
 // and are fully read. To stop the method immediately, close the channels and
 // close the publisher client to ensure that publishing does not block. This
 // may result is some events being discarded.
-func PublishChannels(client beat.Client, cs ...<-chan beat.Event) {
+func PublishChannels(clients map[string]beat.Client, cs ...<-chan beat.Event) {
 	var wg sync.WaitGroup
 
 	// output publishes values from c until c is closed, then calls wg.Done.
 	output := func(c <-chan beat.Event) {
 		defer wg.Done()
+		logger := logp.NewLogger("PublishChannels")
+
+		sink, staticModule := clients[staticModuleSink]
+
 		for event := range c {
-			client.Publish(event)
+			if staticModule {
+				sink.Publish(event)
+				continue
+			}
+
+			v, err := event.Fields.GetValue("metricset.name")
```

Comment:
I think that we should find a way of connecting the metriset with its connector without depending on the content of the event.

Reply:
Refactored. Not relevant anymore.